[CB-261] 반려동물 삭제 기능

dmstmdrbs opened the pull request:
[CB-261]

### 🔨 Jira 태스크

- [CB-349] 반려동물 삭제

### 📐 구현한 내용

- 반려동물 삭제 기능을 구현하였습니다.
- 반려동물 삭제, 정보 업데이트 기능 로직을 묶어 훅으로 분리하였습니다.
  - 다른 컴포넌트에서 재사용 되지 않을 것 같지만 기존에는 각 로직의 코드를 추척하기 어렵다고 느껴져 각 로직 별 코드의 응집도를 높이고자 커스텀 훅으로 분리하였습니다.
   


### 🚧 논의 사항
- 반려동물 삭제 시 상태 경에 따라 페이지를 벗어나기 전에 리렌더링이 발생하게 되어 이미지 src에 접근하려고 해 403 에러가 발생합니다. 사용자 입장에서 사용 상 문제되는 부분은 아니나 해결 방법을 생각해 볼 필요가 있을 것 같습니다.


[CB-261]: https://cocobob.atlassian.net/browse/CB-261?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CB-349]: https://cocobob.atlassian.net/browse/CB-349?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

dmstmdrbs committed:
CB-349 feat: delete pet

반려동물 삭제 기능 추가
- 업데이트, 삭제 기능별 훅으로 로직 분리

dmstmdrbs committed:
CB-349 refactor: delete unused codes

taehong0-0 commented on src/pages/Mypage/pets/[id].tsx:
```diff
@@ -49,6 +53,63 @@ interface IPetEditForm {
+    deleteConfirm,
+    ...response,
+  };
+}
```
위의 로직들을 같은 파일에 둔 이유가 있을까요??
분리해도 어떤 역할을 하는지 알 수 있을 것 같아 분리하는 것도 좋아보입니다 !

dmstmdrbs replied:
다른 파일에서 사용되지는 않을 것 같아서 한 파일에 같이 두었습니다!

dmstmdrbs committed:
CB-373 feat: add new Popup components with useConfirm hooks

새로운 Confirm 팝업 컴포넌트 추가
- 기존의 ToastConfirm을 대체하기 위한 컴포넌트
- useConfirm 훅을 사용하여 리턴값으로 받아온 배열의 첫 번째 원소인 함수를 이용하여 팝업을 open
  - promise를 반환하며, 팝업 확인 시 true, 취소 시 false 반환
	- asyc/await를 꼭 사용해야 함
	- 팝업의 내용을 openPopup의 인자로 넘겨주어야 함

taehong0-0 reviewed (approved):
고생하셨습니다 !

dmstmdrbs committed:
CB-373 refactor: change popup from ToastConfirm to new popup

ToastConfirm 팝업에서 새로운 팝업으로 변경
- 기존의 ToastConfirm 사용 부분을 새로운 팝업으로 대체
- ToastConfirm 팝업 삭제

dmstmdrbs committed:
CB-373 refactor: add useConfirm file

useConfirm 훅을 다른 파일로 분리

dmstmdrbs committed:
CB-375 refactor: modify module import/export

모듈 import/export 관리 방식 변경
- index에서 취합하여 export 시킴.
- 한 곳에서 하위 폴더의 컴포넌트를 모두 가져올 수 있음

dmstmdrbs committed:
Merge pull request #71 from PinkDumbbell/CB-375-refactor/module

[CB-375] 모듈 관리 변경

dmstmdrbs committed:
CB-350 refactor: modify current pet to representative pet

현재 동물을 대표 동물로 변경
- 현재 동물이라는 개념을 없애고 대표 동물로 수정
- currentPet 부분을 대표동물도 바꿔야 함

dmstmdrbs committed:
CB-119 refactor: modify update pet api request body type

반려동물 수정 API 요청 바디 타입 수정

dmstmdrbs committed:
Merge pull request #70 from PinkDumbbell/CB-373-feat/popup

[CB-373] Confirm 팝업 리팩토링
